### PR TITLE
Allow benchmarking optaplanner

### PR DESCRIPTION
**Pull request by hustodemon**
This PR makes use of the [optaplanner benchmarker](https://docs.optaplanner.org/7.30.0.Final/optaplanner-docs/html_single/index.html#benchmarker). This could improve tweaking the optaplanner parameters. More information is contained in the `BENCHMARKING.md`.

Notable changes:
- switch optaplanner to the xml configuration: this is for convenience, since the benchmarker operates on xml files, it's easier to convert the config back and forth between the matcher configuration and the benchmarker configuration.
- tricky part: benchmarker dependency: I didn't want to include the dependency for production builds, so I declared it only in the `test` scope (it basically has the same handling as the `jUnit` library). This is not the clearest approach, but it seems to work (OBS uses `-DskipTests` in the builds, so the `test`-scoped dependencies are not included. 
Tried other approaches too, but they weren't nice either:
  - modularizing the matcher maven project (might be the right way, but it's an overkill)
  - using maven profiles to include the library only when a specified profile is activated (IDE was complaining about missing dependency).

### CODE REVIEW

File: src/main/java/com/suse/matcher/OptaPlanner.java [hunk 1]
```diff
@@ -99,155 +77,17 @@ public OptaPlanner(Assignment unsolved, boolean testing) {
      * @return the solver
      * @param testing true if running as a unit test, false otherwise
      */
-    @SuppressWarnings("rawtypes")
     private Solver initSolver(boolean testing) {
-        // init basic objects
```
Comment (moio) [inline src/main/java/com/suse/matcher/OptaPlanner.java:104]: I will not hide I am a little bit sad to see this code go, it really took a lot of care to make sure it was as readable and self-explaining as possible.

OTOH, I see the need for proper benchmarks in the long run and recognize that as you will likely be the one needing to work on it, it's best for you to own this bit and ultimately decide what to do.

So it goes away... But for a good cause :-)

File: src/main/java/com/suse/matcher/OptaPlanner.java [hunk 2]
```diff
@@ -99,155 +77,17 @@ public OptaPlanner(Assignment unsolved, boolean testing) {
...
-         */
-        TerminationConfig termination = new TerminationConfig();
-        termination.setUnimprovedStepCountLimit(1000);
-        termination.setStepCountLimit(15_000);
-        termination.setHoursSpentLimit(1L);
-        search.setTerminationConfig(termination);
+        SolverConfig config = factory.getSolverConfig();
 
-        /*
-         * Tweak parameters in unit tests, which deal with much less data and need
```
Comment (moio) [inline src/main/java/com/suse/matcher/OptaPlanner.java:239]: Maybe this comment can stay, given the code stays.

File: src/main/resources/com/suse/matcher/config/optaplanner/config.xml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                <entitySelector mimicSelectorRef="entitySelector"/>
+                <valueSelector>
+                    <cacheType>PHASE</cacheType>
+                    <selectionOrder>ORIGINAL</selectionOrder>
+                </valueSelector>
+            </changeMoveSelector>
+        </queuedEntityPlacer>
+    </constructionHeuristic>
+    <localSearch>
+        <!--
```
Comment (moio) [inline src/main/resources/com/suse/matcher/config/optaplanner/config.xml:57]: Indentation seems to be off here

**Comment by hustodemon** [comment]
Thanks for the review. This can be re-opened and improved whenever there is a need for that.

**Pull request closed without merging**